compiler/inferring/type-data.cpp

```diff
@@ -891,30 +856,46 @@ size_t TypeData::get_tuple_max_index() const {
   return subkeys_values.size();
 }
 
+bool TypeData::did_type_data_change_after_tinf_step(const TypeData *before) const {
+  if (ptype_ != before->ptype_ || flags_ != before->flags_) {
+    return true;
+  }
+  if (!class_type_.empty() && class_type_ != before->class_type_) {
+    return true;
+  }
+  if ((anykey_value == nullptr) != (before->anykey_value == nullptr)) {
+    return true;
+  }
+  if (anykey_value != nullptr && anykey_value->did_type_data_change_after_tinf_step(before->anykey_value)) {
+    return true;
+  }
+
+  // most likely we have no subkeys and return false now
+  if (subkeys_values.empty() && before->subkeys_values.empty()) {
+    return false;
+  }
+
+  auto i1 = subkeys_values.begin();
+  auto i2 = before->subkeys_values.begin();
+  auto e1 = subkeys_values.end();
+  auto e2 = before->subkeys_values.end();
+  for (; i1 != e1 && i2 != e2; ++i1, ++i2) {
+    if (i1->second->did_type_data_change_after_tinf_step(i2->second)) {
```

Comment:
Shouldn't we check key here?

Reply:
No, we shouldn't. Because even if keys were swapped, we should return that this was changed

Reply:
Sounds reasonable. I'll add `i1->first != i2->first || ` condition